src/librustdoc/html/markdown.rs

```diff
@@ -1064,11 +1065,6 @@ fn markdown_summary_with_limit(
     let mut text_length = 0;
     let mut stopped_early = false;
 
-    fn push(s: &mut String, text_length: &mut usize, text: &str) {
```

Comment:
Why removing this function? It allowed us to automatically update`text_length`, which seems much better than doing it "manually" every time...

Reply:
Because the escaped HTML will be longer than the original text; e.g., `Rust's "type system"` would turn into something like `Rust&#39; &#34;type system&#34;`, which is 12 characters longer than how it will be displayed. I'm planning to refactor this code to separate the length-limiting and rendering parts soon, but for now I think it's best to just increase the length manually.

Reply:
In 2 out of the 3 cases, you use `Escape` and the last one is simply a whistespace so it can be escaped without issues as well. Then why not change the `push` to this:

```rust
fn push(s: &mut String, text_length: &mut usize, text: &str) {
    write!(s, "{}", Escape(text)).unwrap();
    *text_length += text.len();
}
```

?

Reply:
That's a good point; I think I didn't do that because at first I was just escaping code :)

I'll change it to do that.